php/includes/language.php

```diff
@@ -91,3 +91,62 @@ function FN_(string $singular, string $plural, ...$replacePairs)
 		wrap_array_keys_in_braces($replacePairs),
 	);
 }
+
+// custom translation framework
+include('../includes/static_translation_table.php');
+
+// Translate a key according to $acceptedLanguage, given its translation key
+// e.g. mkpc_get_translated_string("kHELLO"), $acceptedLanguage = "fr"
+/// will yield "Salut".
+// If $forceLanguage is specified, it will override $acceptedLanguage.
+function mkpc_get_translated_string(string $key, string $forceLanguage = "") {
+	global $acceptedLanguage;
+	$language = $acceptedLanguage;
+	if (!empty($forceLanguage)) {
+		$language = $language . "#" . $forceLanguage;
+	}
+	if (array_key_exists($key, TRANSLATION_TABLE)) {
+		$translation = TRANSLATION_TABLE[$key];
+		if (array_key_exists($language, $translation)) {
+			return $translation[$language];
+		} else if (array_key_exists("en", $translation)) {
+			// Default to english
+			trigger_error("Translation error: key '" . $key . "' could not be translated to '" . $language . "' (missing language?). Used default language instead.");
+			return $translation["en"];
+		} else {
+			trigger_error("Translation error: key '" . $key . "' could not be translated (missing default language?). Used key instead.");
+			return $key;
+		}
+	} else {
+		trigger_error("Translation error: key '" . $key . "' could not be translated (missing key?). Used key instead.");
+		return $key;
+	}
+}
+// Shorthand for the function above
+function t()
+{
+	return call_user_func_array("mkpc_get_translated_string", func_get_args());
+}
+
+// Same as above, but will format a string, à la python.
+// E.g. Ft(kHELLO_PARAM_NAME, name: "Wargor")
+// will yield "Hello Wargor!"
+function Ft(string $msgid, ...$replacePairs)
+{
+	return strtr(mkpc_get_translated_string($msgid), wrap_array_keys_in_braces($replacePairs));
+}
+
+// Same as above, but has special handling for plurals.
+// E.g. FNt("kYOU_HAVE_MESSAGE_PARAM_COUNT", count: 0)
+// will yield "You have 0 new messages", while
+// FNt("kYOU_HAVE_MESSAGE_PARAM_COUNT", count: 0)
+// will yield "You have 1 new message".
+// There must be a parameter called "count". Other parameters can be added,
+// and will format as usual.
+function FNt(string $msgid, ...$replacePairs) {
+	$languageContext = "other";
+	if ($replacePairs["count"] != 0) {
```

Comment:
Why condition is `!= 0` here? This doesn't seem to apply the plural when count >= 2